Project auth

camillalof opened the pull request:
Link to my deployed project: https://elegant-tesla-808cf0.netlify.com/

My reflections on how this week's project turned out:

This was not an easy one. :) The week started of ok but as the week went by it felt like we where in the dark sort of. I still dont really understand the conection between frontend and backend, and what it is that creates the authorization in the frontend code. It would of been so great if there had been a fourth video week 20 explaining exactly that. Would that be possible to add one to canvas?  I have a 1:1 booked where I want to go trough this but in the project. But I really feel like this also would be such a nice project to have in the portfolio, and I would feel more confident knowing that I understand the logic of authorization in the frontend. :)

Things I'd like to have clarified or explained in more detail:

This code is a mix of mine and Petras code from the start, but we have asked for alot of help along the way. So we added Jennies code, then Vans code, then we added code from Teresa and then Nasim helped us so we added her code :D My backend did work from the start but with all the codes front to back it only says "cannot get". So I would love to get some feedback on why I get that error message. Right now my frontend-page is connected to Petras backend to work, but my original code is in git-hub.

The page is not working deployed, I get an error message that says that it cant take me to the member page and I have had problems saving a user. It would be great to get feedback on that.

I would also love to get an explanation on what part of the code saves the token and let you go to the next page (and se the "secrets"). Does that happen in login or only in the member page. If it happens in both what part of the code keeps track on that. And what part of the code stops that to happen (like for example when you log out? Do we have that in our code or is there a function that we should write?

Thanks alot!

camillalof commented:
This is the version that both me and Petra is making a pull request on. Sorry I have been thinking so much today im a little confused. :)  But this is the pull request for both of us. Me and Petra was writing together on two computers. 

So we have the same problem in the frontend like I described. This is the repo from her computer just in case https://github.com/petrabrinck/project-auth. But they are almost the same, if something is different its a detail or something. So the frontend problems are the same for both of us. :) The backend is my problem. 

Ok I might confuse you guys now. Ask if you wonder something! :)

puj commented on backend/server.js:
```diff
@@ -15,13 +37,66 @@ const port = process.env.PORT || 8080
+    res.status(400).json({ message: 'could not find user', errors: err.errors })
+  }
 })
+}  
```
This extra `}` is actually closing the function on line https://github.com/Technigo/project-auth/pull/12/files#diff-5784859bc20a9558df9dfce578b2f673R43

Line 43: ` const authenticateUser = async (req, res, next) => { `

Which means ALL of your endpoints are not defined until this function is called.  

This is where formatting can be SOO helpful.  You could see that the functions weren't exactly on the left edge of the editor, so they were inside of a different code block.  

Once this was fixed.  I was able to create a user.

camillalof replied:
Thank you! I understand that I have a } on the wrong line. But where is it supposed to be? line 59? :)

camillalof replied:
Or was it some other line? :)

puj replied:
Yeah, it was supposed to be on 59 to close that function.  Once you change that, the other endpoints are defined outside of that function and work properly 👍

puj commented on backend/server.js:
```diff
@@ -15,13 +37,66 @@ const port = process.env.PORT || 8080
+// login user
+app.post('/sessions', async (req, res) => {
+  try {
+    const { name, password } = req.body
```
Another issue is here:  You are expecting name from the `req.body`, but on 
https://github.com/Technigo/project-auth/pull/12/files#diff-3c712706d81b44f9660bc1b5cf8b1fa9R17

You are sending `{email, password}` from your frontend so mongo will not find the correct user when looking for a matching email with the name value

camillalof replied:
Fixed :)

puj commented on backend/server.js:
```diff
@@ -15,13 +37,66 @@ const port = process.env.PORT || 8080
+app.post('/sessions', async (req, res) => {
+  try {
+    const { name, password } = req.body
+    const user = await User.findOne({ name })
```
```javascript
    const { email, password } = req.body
    const user = await User.findOne({ email: email })
```
seems to help

camillalof replied:
fixed :)

camillalof commented:
Hello Van, I fixed the frontend code. Regarding the backend code me and Petra went with the code on her backend for the project (the backend code here: https://github.com/petrabrinck/project-auth) I have fixed some things in the backend on this page but that was more an extra question since we worked together and Petras backend worked. So now we have a working project :) Will this mean that we both are approved on this assignment?
